RockPaperScissors.c

```diff
@@ -5,68 +5,78 @@
 
 int main(void)
 {
-    
     // Initialize random number generator
     srand(time(NULL));
-    int n;
-    int x = 0;
-    int Rock = 1; 
-    int Paper = 2; 
-    int Scissors = 3; 
+
+    // Allocate space for variables
+    int userChoice;
+    int aiChoice;
+    int round = 0;
+
+    // Initialize constants
+    const int Rock = 1; 
+    const int Paper = 2; 
+    const int Scissors = 3; 
     
     do 
     {
         printf ("Your turn! \n");
         do
         {
-            n = get_int ("Rock = 1, Paper = 2, Scissors = 3. Please enter a number to play: \n");
+            userChoice = get_int ("Rock = 1, Paper = 2, Scissors = 3. Please enter a number to play: \n");
         }
-        while (n < 1 || n > 3);
+        while (userChoice < Rock || userChoice > Scissors);
     
-        if (n == 1)
+        if (userChoice == Rock)
         {
             printf ("You picked Rock. Now, it's my turn! \n");
         }
-        else if (n == 2)
+        else if (userChoice == Paper)
         {
             printf ("You picked Paper. Now, it's my turn! \n");
         }
-        else
+        else if (userChoice == Scissors)
         {
             printf ("You picked Scissors. Now, it's my turn! \n");
         }
+        else
+        {
+           // This shouldn't normally happen. But if it does, the computer will warn the user about it.
+           // This will save debugging nightmares in the future.
+           printf("ERROR: The computer accepted a value other than 1, 2, 3. This should not happen. Please check business logic. Exiting."); 
+           exit(-1) 
+        }
     
         printf ("Rolling the dice .... \n");
     
-        int a = rand() % 3 + 1;
+        aiChoice = rand() % 3 + 1;
```

Comment:
To be [consistent](http://queirozf.com/entries/on-coding-consistency), I moved `aiChoice` up to the top with the other variable declarations.

#### HEADS UP: Because I have not tested the code, this change may cause functional issues.